ci_tests/system.py

```diff
@@ -111,153 +122,204 @@ def test_execution_with_upload(self):
-
-            response, status_code = self.get_archive_experiments_by_page(self.BASE_URL, self.demo_id)
+                response = self.run_demo_with_uploaded_blob(
+                    self.demo_id, blob, width, height
+                )
+            run_status = response.get("status")
+
+            response, status_code = self.get_archive_experiments_by_page(
+                self.BASE_URL, self.demo_id
+            )
             archive_status = status_code
-            experiments = response.get('experiments')
-            exp_meta = response.get('meta_info')
+            experiments = response.get("experiments")
+            exp_meta = response.get("meta_info")
             try:
-                n_experiments = exp_meta.get('number_of_experiments')
+                n_experiments = exp_meta.get("number_of_experiments")
             except Exception:
                 pass
 
             # Delete demo
-            response = self.delete_demo(self.demo_id)
-            delete_demo_status = response.get('status')
+            response, _ = self.delete_demo(self.demo_id)
+            delete_demo_status = response.get("status")
 
         finally:
-            self.assertEqual(create_demo_status, 'OK')
-            self.assertEqual(add_ddl_status, 'OK')
-            self.assertEqual(demo_extras_status, 'OK')
-            self.assertEqual(run_status, 'OK')
+            self.assertEqual(create_demo_status, "OK")
+            self.assertEqual(add_ddl_status, "OK")
+            self.assertEqual(demo_extras_status, "OK")
+            self.assertEqual(run_status, "OK")
             self.assertEqual(archive_status, 200)
             self.assertTrue(isinstance(experiments, list))
             self.assertTrue(n_experiments > 0)
-            self.assertEqual(delete_demo_status, 'OK')
+            self.assertEqual(delete_demo_status, "OK")
 
     #####################
     #       TOOLS       #
     #####################
 
-    def post(self, module, service, params=None, data=None, files=None, servicejson=None):
-        url = '{}/api/{}/{}'.format(self.BASE_URL, module, service)
-        return requests.post(url, params=params, data=data, files=files, json=servicejson)
+    def post(self, module, method, service, **kwargs):
+        """
+        post
+        """
+        url = f"{self.BASE_URL}/api/{module}/{service}"
+        if method == "get":
+            return requests.get(url)
+        elif method == "put":
+            return requests.put(url, **kwargs)
+        elif method == "post":
+            return requests.post(url, **kwargs)
+        elif method == "delete":
+            return requests.delete(url, **kwargs)
+        else:
+            assert False
 
     def check_status(self, response, error_msg):
         """
         If the response is not OK it will log the error
         """
-        if 'status' not in response or response['status'] == 'KO':
+        if "status" not in response or response["status"] == "KO":
```

Comment:
Check for  != "OK" and no for  == "KO". I know this code was already there before your refactoring, but it'd a good moment to fix this, even though.